File: 997/997.py
from typing import List
def findJudge(n: int, trust: List[List[int]]):
    trusts = [0 for i in range(n)]
    for pair in trust:
        trusts[pair[1] -1] += 1
        trusts[pair[0] -1] -= 1
    if n-1 in trusts:
        judge = trusts.index(n-1) +1
    else: judge = -1
    return judge

if __name__ == "__main__":
    print(findJudge(3,[[1,3],[2,3],[3,1]]))

File: 997/997.cpp
#include<iostream>
#include <vector>
using namespace std;

int findJudge(int n, vector<vector<int>>& trust) {
        vector<int>trustScore(n+1,0);
        for(auto person:trust){
            trustScore[person[0]]--;
            trustScore[person[1]]++;
        }
        for(int i=1;i<=n;i++){
            if(trustScore[i]==n-1){
                return i;
            }
        }
        return -1;
    }

int main(){
	int n = 3;
	vector<vector<int>> trust = {{1,3},{2,3},{3,1}};
    cout << findJudge(n, trust) << endl;
    return 1;
}